### ISSUE TITLE
[Feature]: Make tags auto-appending configurable

### ISSUE THREAD

**Issue by vitalets** [microsoft/playwright#33947]
### 🚀 Feature Request

After https://github.com/microsoft/playwright/pull/32930 tags are automatically appended to test titles in `list` and `line` reporters.

Could you still consider making this behavior configurable?
Imho one of the benefits of tags annotations is cleaner output. In case of many tags appended to test title, it may look cluttered and unexpected to users.

Here is the report I've got for playwright-bdd https://github.com/vitalets/playwright-bdd/issues/251

### Example

I'd suggest to keep originally proposed `appendTags` option (with default to `true`):
```ts
reporter: [
  ['list', { appendTags: false }]
],
```

### Motivation

- cleaner reporter output
- no unexpected modifications of user's input

**Comment by vitalets** [comment]
> How do you want these tags to be represented if not in the reporter? If you don't want them to be printed maybe don't add them as tags?

I think HTML reporter is a great place to show all tags. Visual chips are nice and clickable. 
Showing all tags in terminal appended to the title is not very useful for my workflow.

**Comment by dimkin-eu** [comment]
+1 to issue. exposing tags in console output in CI clutters test output ( we have 10+ tags for some tests )

tags weren't exposed in version 1.46 for sure and appeared in 1.49 at least

**Comment by vitalets** [comment]
The same problem, submitted earlier: https://github.com/microsoft/playwright/issues/17375

**Comment by Az8th** [comment]
I agree that this would be greatly appreciated, for all the aforementioned use cases, and also as it would enhance visibility in CI for users that don't rely on the dot reporter.

My tests are executed with tools and within contexts that don't permit to show status for each test as they go unless the whole suite is performed. And for certain urgent situations, like testing hotfixes before they are sent to production, I directly look to the pipeline output as tests run, and I too have some tests that have tags that may not be useful to display (as they are purely used for technical perspectives)
 
However, it should be taken into account that tags can be added from both tags attribute, and test title.
What would be the expected behaviour here ? Remove them all ?
Because I tend to use key words contained in the test titlle as tags, and to declare other ones with the tags attribute.
Example : `@Login : I can authenticate with a @certificate`,
Would it display as ` : I can authenticate with a` ?

If so, it would be interesting to add the ability to chose to hide them depending on their source
Something like :
```typescript
reporter: [
  ['list', { hideTagsFrom: 'all' }] //'title' or 'tags'
],
```

**Comment by Az8th** [comment]
Actually, my above solution is quite inelegant, as it would force to use a source rather than another depending on either you would want to hide the tag or not, and a filter would require to change configuration file every time.

Why not instead give tags the same treatment as annotations by hiding those prefixed with an `_` ? Plus, it would also profit for all reports, especially HTML. If this needs a separate issue, tell me and I would fill it, but I think both features should be packed together to ensure a better flexibility

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
